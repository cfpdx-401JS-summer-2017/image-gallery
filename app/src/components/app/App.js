import React, { Component } from 'react';
import { BrowserRouter as Router,
        Switch,
        Redirect,
        Route,
        Link
} from 'react-router-dom';
import Home from '../home/Home';
import About from '../about/About';
import './App.css';
import { images, addImage, removeImage } from '../images/images';
import ViewSelector from '../images/ViewSelector';
import AddImage from '../images/AddImage';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      images: images
    }
  }

  addImage = (title, description, url) => {
    const img = {title, description, url};
    const oldImages = this.state.images;
    console.log('img', img);
    this.setState({
      images: addImage(oldImages, img)
    })
  }

  removeImage = image => {
    const oldImages = this.state.images;
    this.setState({
      images: removeImage(oldImages, image)
    })

  }

  render() {

    return (

      <Router>
      <div className="App">
          <Switch>
          <Route exact path="/" component={Home}/>
          <Route exact path="/about" component={About}/>
          <Route path="/images" render={({ match, location}) => {
            return <div>
              <Link to="/home"> <button>Home</button> </Link>
              <Link to="/about"> <button>About</button> </Link>
              <ViewSelector images={this.state.images} onRemove = {this.removeImage} />
              <AddImage onAdd={this.addImage}/> 
              </div>
          }}/>
          <Redirect to="/"/>
          </Switch>
      </div>
      </Router>
    );
  }
}

export default App;

