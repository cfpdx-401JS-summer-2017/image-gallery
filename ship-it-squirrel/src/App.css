.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.list-view {
  display: block;
  text-align: left;
}

.thumbnail-view div {
  display: inline-block;
  text-align: center;
}