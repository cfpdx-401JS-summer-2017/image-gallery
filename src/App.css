header {
  background: #232;
  border: 1px solid #232;
}
header h1 {
  text-align: center;
  color: #fff;
}
nav {
  background: #252;
  padding: 1em 0;
  text-align: center;
}
nav button {
  padding: .5em 1em;
  background: #cfc;
  border: none;
  font-size: 1em;
  margin-right: 1em;
}
nav button:last-child {margin-right: 0;}

section {margin: 2em;}
section ul {
  list-style: none;
  margin-bottom: 2em;
}
.thumbView {
  float: left;
  height: 150px;
  width: 125px;
  border: 1px solid #999;
  text-align: center;
  padding: 10px;
  margin-right: 10px;
}
.thumbView img {
  max-width: 100px;
  max-height: 100px;
}
.thumbView p { font-size: .8em; }
.clear:after {
  content: "";
  display: table;
  clear: both;
}

.galleryView img {
  max-width: 600px;
}

form input {
  margin: 0 1.5em 0 1em;
}
form button {
  border: none;
  background: #252;
  color: #fff;
  font-size: .8em;
  padding: .5em 1em;
}