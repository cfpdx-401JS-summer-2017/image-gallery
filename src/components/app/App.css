header {
  background: #232;
}
header h1 {
  text-align: center;
  color: #fff;
  padding: .75em 0;
  font-size: 2em;
}
nav ul li {display: inline-block;}
nav ul li a {text-decoration: none; color: #232;}
nav#mainNav {
  background: #252;
  padding: 1.5em 0;
  text-align: center;
}
nav#mainNav ul li a {
  padding: .5em 1em;
  background: #cfc;
  border: none;
  font-size: 1em;
  margin-left: 2em;
  color: #232;
}
nav#mainNav ul li:first-child a {margin-left: 0;}
nav#mainNav ul li a:hover {background: #ffc;}

nav#subNav {
  margin: 2em 0 2em 2em;
}
nav#subNav ul li a {
  font-size: .9em;
  margin-right: 1em;
  border: 1px solid #232;
  padding: .25em .5em;
}

section {margin: 2em;}
section h2 {
  font-weight: bold;
  font-size: 1.5em;
  margin-bottom: 1em;
}
section p {margin-bottom: 1em;}
section a {text-decoration: none; color: #282;}
section ul {
  list-style: none;
  margin-bottom: 2em;
}
.thumbView {
  float: left;
  height: 125px;
  width: 125px;
  border: 1px solid #999;
  text-align: center;
  padding: 20px 10px;
  margin-right: 20px;
  position: relative;
}
.thumbView img {
  max-width: 100px;
  max-height: 100px;
}
.thumbView p { font-size: .8em; }
.thumbView button, .galleryView button {
  position: absolute;
  top: -5px;
  left: -5px;
  font-weight: bold;
  background: #c00;
  border: none;
  color: #fff;
  box-shadow: -1px 1px 1px rgba(0,0,0,.5);
}
.clear:after {
  content: "";
  display: table;
  clear: both;
}

.galleryView {
  width: 600px;
  position: relative;
  box-shadow: 0 0 4px 2px rgba(0,0,0,.2);
  padding: 10px;
}
.galleryView h3 {
  font-weight: bold;
  text-align: center;
  margin: 10px 0 5px;
}
.galleryView p {
  font-size: .8em;
  font-style: italic;
  text-align: center;
}
.galleryView img {
  width: 600px;
  border: 1px solid #ccc;
}
.galleryView button {font-size: 1em;}

#galleryNav {
  margin-bottom: 2em;
  text-align: center;
  width: 620px;
}
#galleryNav button {
  font-size: 1em;
}
#bunnyCounter {
  margin-top: 10px;
  font-size: .8em;
}

form input {
  margin: 0 1.5em 0 1em;
}
form button {
  border: none;
  background: #252;
  color: #fff;
  font-size: .8em;
  padding: .5em 1em;
}