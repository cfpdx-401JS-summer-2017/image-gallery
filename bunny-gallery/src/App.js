import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { bootstrap } from './image';

const images = [
  { title: 'cute bunny', description: 'very cute bunny', url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg' },
  { title: 'another cute bunny', description: 'very, very cute bunny', url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg' },
  { title: 'fairly cute bunny', description: 'pretty much a cute bunny', url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg' },
];

class ImageList extends Component {

  constructor(props) {
    super(props);
    this.state = {
      view: views[0]
    }
  }

  render() {
    return (
      <ul>
        {images.map((image, index) => {
          return (<li key={index}> <a href={image.url}> {image.title}</a>
            <p>{image.description}</p>
          </li>
          )
        })}
      </ul>
    )
  }
}
class ImageThumbnail extends Component {

  constructor(props) {
    super(props);
    this.state = {
      view: views[1]
    }
  }

  render() {
    return (
      <div>
        {images.map((image, index) => {
          return (
            <div>
              <img key={index} src={image.url} width ="200"/>
              <p>{image.title}</p>
              <p>{image.description}</p>
            </div>
          )
        })};
        </div>
    )
  }
}
class ImageGallery extends Component {

  constructor(props) {
    super(props);
    this.state = {
      view: views[2]
    }
  }

  render() {
    return (
      <div>
        {images.map((image, index) => {
          return (
            <div>
              <button>Previous</button>
              <img key={index} src={image.url} width ="200"/>
              <p>{image.title}</p>
              <p>{image.description}</p>
              <button>Next</button>
            </div>
          )
        })};
      </div>
    )
  }
}


const View = {
  list: ImageList,
  thumbnail: ImageThumbnail,
  gallery: ImageGallery
}

const views = Object.keys(View);

class App extends Component {

  constructor() {
    super();
    this.state = {
      // search: 'bunny',
      view: views[1],
      views: views,
      images: bootstrap()
    }
  }

  render() {
    const { views, view, images } = this.state;
    const ImageView = View[view];


    return (
      <div className="App">
        <div>
          {views.map(v => (
            <button key={v} onClick={() => this.setState({ view: v })}>
              {v}</button>))}
          {/* <div><p>{this.state.images}</p></div> */}
          <ImageView />
        </div>
      </div>
    );
  }
}

export default App;

