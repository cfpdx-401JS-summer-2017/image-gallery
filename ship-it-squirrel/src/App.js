import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import EmojiList from './views/list'
import EmojiThumbnail from './views/thumbnail'
import EmojiGallery from './views/gallery'
import ViewOptions from './views/options'

const views = {
  list: EmojiList,
  thumbnail: EmojiThumbnail,
  gallery: EmojiGallery,
};

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      view: 'list',
      emojis: [
        {
          title: 'ship it squirrel',
          description: 'GitHub\'s :shipit: emoji is a mobster squirrel used, typically in PR\'s, to communicate the code is finished and ready for production. Ship it!',
          url: 'https://qph.ec.quoracdn.net/main-qimg-c8781a4bb1f17e330b50cb35f851da05-c'
        },{
          title: 'hamburger',
          description: 'A hamburger or burger is a sandwich consisting of one or more cooked patties of ground meat, usually beef, placed inside a sliced bread roll or bun. The patty may be pan fried, barbecued, or flame broiled. Hamburgers are often served with cheese, lettuce, tomato, bacon, onion, pickles, or chiles; condiments such as mustard, mayonnaise, ketchup, relish, or "special sauce"; and are frequently placed on sesame seed buns. A hamburger topped with cheese is called a cheeseburger.',
          url: 'https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/hamburger_1f354.png',
        },{
          title: 'unicorn',
          description: 'The unicorn is a legendary creature that has been described since antiquity as a beast with a single large, pointed, spiraling horn projecting from its forehead. The unicorn was depicted in ancient seals of the Indus Valley Civilization and was mentioned by the ancient Greeks in accounts of natural history by various writers, including Ctesias, Strabo, Pliny the Younger, and Aelian.[1] The Bible also describes an animal, the re\'em, which some versions translate as unicorn.',
          url: 'https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/unicorn-face_1f984.png',
        }
      ]
    }

    this.handleViewChange = this.handleViewChange.bind(this);
  }

  handleViewChange(view) {
    this.setState({ view });
  }

  render() {

    let CurrentView = views[this.state.view];

    return (
      <div className="main">
        <div>
          <h1>Christy's Image Gallery</h1>
          <ViewOptions onClick={this.handleViewChange}/>
        </div>

        <CurrentView emojis={this.state.emojis}/>

      </div>
    );
  }
}

export default App;
