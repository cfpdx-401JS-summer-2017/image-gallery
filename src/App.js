import React, { Component } from 'react';
import {Switch, Link, Route, Redirect} from 'react-router-dom';
import Home from './Home';
import About from './About';
import Images from './Images';
import './App.css';

class App extends Component {
  
  render() {
    return (
      <main className="App">
        <header className="App-header">
          <h1>Steve Pics Dot Com!</h1>
          <nav>
            <Link to="/">Home</Link>
            <Link to="/about">About</Link>
            <Link to="/images">Steve Pics</Link>
          </nav>
        </header>
        <Switch>
          <Route exact path="/" component={Home}/>
          <Route path="/about" component={About}/>
          <Route path="/images" component={Images}/>
          <Redirect to="/"/>
        </Switch>
      </main>
    );
  }
}

export default App;
