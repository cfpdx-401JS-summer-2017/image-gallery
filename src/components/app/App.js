import React, { Component } from 'react';
import './App.css';
import Nav from '../Nav';

import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';
import Home from '../../routes/Home';
import About from '../../routes/About';
import { Images } from '../../routes/Images';

class App extends Component {
  
  render() {
    return (
      <Router>
        <div>
          <header>
            <h1>Cute Bunnies: The App</h1>
            <Nav />
          </header>
          <main>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route path="/images" component={Images} />
              <Route path="/about" component={About} />
              <Redirect to="/" />
            </Switch>
          </main>
        </div>
      </Router>
    );
  }
}

export default App;

