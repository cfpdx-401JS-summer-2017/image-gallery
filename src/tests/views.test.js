import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';

import EmojiList from '../views/list';
import EmojiThumbnail from '../views/thumbnail';
import EmojiGallery from '../views/gallery';

describe('views components', () => {

    it('renders the LIST view  without crashing', () => {
        const testEmojis = [];
        const div = document.createElement('div');
        ReactDOM.render(<EmojiList emojis={testEmojis}/>, div);
    });

    it('renders the THUMBNAIL view  without crashing', () => {
        const testEmojis = [];
        const div = document.createElement('div');
        ReactDOM.render(<EmojiThumbnail emojis={testEmojis}/>, div);
    });

    it('renders the GALLERY view  without crashing', () => {
        const testEmojis = [];
        const testEmojiIndex = 1;
        const div = document.createElement('div');
        ReactDOM.render(<EmojiGallery emojis={testEmojis} emojiIndex={testEmojiIndex} />, div);
    });

});
