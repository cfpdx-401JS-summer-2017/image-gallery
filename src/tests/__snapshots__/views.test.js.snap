// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views components gallery renders a snapshot 1`] = `
<div
  className="gallery-view"
>
  <div>
    <img
      alt="syringe"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/syringe_1f489.png"
    />
    <h2>
      syringe
    </h2>
    <p>
      nurses poke you with it
    </p>
  </div>
  <button
    className="gallery-nav-button"
    onClick={[Function]}
  >
     Back 
  </button>
  <button
    className="gallery-nav-button"
    onClick={[Function]}
  >
     Next 
  </button>
</div>
`;

exports[`views components list renders a snapshot 1`] = `
<div
  className="list-view"
>
  <div>
    <h2>
      syringe
    </h2>
    <img
      alt="syringe"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/syringe_1f489.png"
    />
    <p>
      nurses poke you with it
    </p>
  </div>
  <div>
    <h2>
      sushi
    </h2>
    <img
      alt="sushi"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/sushi_1f363.png"
    />
    <p>
      super yummy!
    </p>
  </div>
  <div>
    <h2>
      pile of poo
    </h2>
    <img
      alt="pile of poo"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/pile-of-poo_1f4a9.png"
    />
    <p>
      what I look like doing snapshot testing
    </p>
  </div>
</div>
`;

exports[`views components thumbnail renders a snapshot 1`] = `
<div
  className="thumbnail-view"
>
  <div>
    <img
      alt="syringe"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/syringe_1f489.png"
    />
    <p>
      syringe
    </p>
  </div>
  <div>
    <img
      alt="sushi"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/sushi_1f363.png"
    />
    <p>
      sushi
    </p>
  </div>
  <div>
    <img
      alt="pile of poo"
      src="https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/96/pile-of-poo_1f4a9.png"
    />
    <p>
      pile of poo
    </p>
  </div>
</div>
`;
